ui/lib/core/addon/components/info-table-item-array.js

```diff
@@ -15,63 +14,81 @@ import { isWildcardString } from 'vault/helpers/is-wildcard-string';
+ * @queryParam="role"
+ * @backend="transform"
+ * viewAll="roles">
  * ```
  *
  * @param displayArray=null {array} - This array of data to be displayed.  If there are more than 10 items in the array only five will show and a count of the other number in the array will show.
- * @param [isLink=true] {Boolean} - Indicates if the item should contain a link-to component.  Only setup for arrays, but this could be changed if needed.
- * @param [modelType=null] {string} - Tells what model you want data for the allOptions to be returned from.  Used in conjunction with the the isLink.
+ * @param [isLink] {Boolean} - Indicates if the item should contain a link-to component.  Only setup for arrays, but this could be changed if needed.
+ * @param [rootRoute="vault.cluster.secrets.backend.list-root"] - {string} - Tells what route the link should go to when selecting "view all".
+ * @param [itemRoute=vault.cluster.secrets.backend.show] - {string} - Tells what route the link should go to when selecting the individual item.
+ * @param [modelType] {string} - Tells which model you want data for the allOptions to be returned from.  Used in conjunction with the the isLink.
  * @param [wildcardLabel] {String} - when you want the component to return a count on the model for options returned when using a wildcard you must provide a label of the count e.g. role.  Should be singular.
  * @param [queryParam] {String} - If you want to specific a tab for the View All XX to display to.  Ex: role
  * @param [backend] {String} - To specify which backend to point the link to.
  * @param [viewAll] {String} - Specify the word at the end of the link View all xx.
  */
-export default Component.extend({
-  layout,
-  'data-test-info-table-item-array': true,
-  attributeBindings: ['data-test-info-table-item-array'],
-  allOptions: null,
-  displayArray: null,
-  wildcardInDisplayArray: false,
-  store: service(),
-  displayArrayAmended: computed('displayArray', function () {
+export default class InfoTableItemArray extends Component {
+  @tracked allOptions = null;
+  @tracked wildcardInDisplayArray = false;
+  @service store;
+
+  get rootRoute() {
+    return this.args.rootRoute || 'vault.cluster.secrets.backend.list-root';
+  }
+
+  get itemRoute() {
+    return this.args.itemRoute || 'vault.cluster.secrets.backend.show';
+  }
+
+  get displayArray() {
+    return this.args.displayArray || null;
+  }
+
+  get displayArrayAmended() {
     let { displayArray } = this;
-    if (!displayArray) {
-      return;
-    }
+    if (!displayArray) return null;
     if (displayArray.length >= 10) {
       // if array greater than 10 in length only display the first 5
       displayArray = displayArray.slice(0, 5);
     }
-
     return displayArray;
-  }),
+  }
 
-  checkWildcardInArray: task(function* () {
+  @task *checkWildcardInArray() {
     if (!this.displayArray) {
       return;
     }
     let filteredArray = yield this.displayArray.filter((item) => isWildcardString(item));
```

Comment:
since `isWildcardString` is a helper - do we have to pass it inside array brackets? Just remember syntax being strange for hbs helpers - but maybe I'm thinking about this wrong here!

Reply:
Good question. Double checked, it seems to handle it fine, I _think_ because the helper uses `isWildcardString([string])`

Reply:
ah yeah! then it destructures it from an array in the function itself